File: calendar2_TokenRing_Algorithm/HPN JAVA TOKEN RING CODE/HpnCalendarTools/src/hpn/console/file/FileIO.java
package hpn.console.file;

import java.io.IOException;
import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

public class FileIO {
	  private String filePath;
	  final static Charset ENCODING = StandardCharsets.UTF_8;
	 
	  public FileIO(String filePath) 
	  {
		super();
		this.filePath = filePath;
	  }

	  public List<String> readFile() throws IOException 
	  {
		  if(this.filePath== null)
			  throw new IllegalStateException("The file path has not assigned yet.");
		    Path path = Paths.get(this.filePath);
		    if(Files.exists(path))
		    	return Files.readAllLines(path, ENCODING);
		    else
		    {
		    	Files.createFile(path);
		    	return null;
		    }
	  }
		  
	  public void writeFile(List<String> allLines) throws IOException 
	  {
		    Path path = Paths.get(this.filePath);
		    Files.write(path, allLines, ENCODING);
	  }

}

File: calendar1_R&A_Algorithm/HPN C# R&A CODE/HpnCalendarTools/HpnCalendarTools/hpn/console/file/FileIO.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;

namespace hpn.console.file
{
    public class FileIO
    {
          private String filePath;
	      
	      public FileIO(String filePath) 
	      {
		    this.filePath = filePath;
	      }

	      public List<String> readFile()
	      {
		        if(this.filePath== null)
			      throw new System.NullReferenceException("The file path has not assigned yet.");

		        if(File.Exists(this.filePath))
		    	{
                    List<string> lines = File.ReadLines(this.filePath).ToList();
                    return lines;
                }
		        else
		        {
		    	    StreamWriter stream = File.CreateText(this.filePath);
                    stream.Close();
		    	    return null;
		        }
	      }
		  
	      public void writeFile(List<String> allLines) 
	      {
              File.WriteAllLines(this.filePath, allLines, System.Text.Encoding.UTF8);
	      }
    }
}
